lib/src/annotations.dart

```diff
@@ -3,9 +3,19 @@ class Entity {
   const Entity({this.uid = null});
 }
 
+/**
+ * A dart int value can map to different OBXPropertyTypes,
+ * e.g. Short (Int16), Int (Int32), Long (Int64), all signed values.
+ * Also a dart double can also map to e.g. Float and Double
+ *
+ * Property allows the mapping to be specific. The defaults are
+ * e.g. Int -> Int64, double -> Float64, bool -> Bool.
+ *
+ * Use OBXPropertyType and OBXPropertyFlag values, resp. for type and flag.
+ */
 class Property {
-  final int uid;
-  const Property({this.uid = null});
+  final int uid, type, flag;
```

Comment:
Do we really need flag annotations?

Reply:
Also I'm not sure I understand why type is necessary for this PR. Maybe it was before the code generator changes ("id generation")? If it's not necessary/used, I'd suggest reverting those changes for now

Reply:
No it's new. The `flag` is there to open up support for all the flags like uniqueness constraints etc. My reasoning: Instead of creating a new annotation for each property, one annotation should cover them all.

In the case of the `type`, I can imagine some users who don't need `Int64` for each int. This allows for some specificity.

Reply:
But you're right this, should be outside of this PR.

Reply:
this would require further discussion so I'm not sure about merging it as part of this PR